radix-transactions/src/model/execution/executable_transaction.rs

```diff
@@ -76,20 +62,41 @@ impl ExecutableTransaction {
 
         Self {
             context,
-            intents: ExecutableIntents::V1(ExecutableIntentV1 {
+            intents: vec![ExecutableIntent {
                 encoded_instructions: encoded_instructions_v1,
-                references: Rc::new(references),
+                references,
                 blobs,
                 auth_zone_init,
-            }),
+                children_intent_indices: vec![],
+            }],
         }
     }
 
-    pub fn new_v2(intents: Vec<ExecutableIntentV2>, context: ExecutionContext) -> Self {
-        Self {
-            context,
-            intents: ExecutableIntents::V2(intents),
+    pub fn new_v2(mut intents: Vec<ExecutableIntent>, context: ExecutionContext) -> Self {
+        for intent in &mut intents {
+            for proof in &intent.auth_zone_init.initial_non_fungible_id_proofs {
+                intent
+                    .references
+                    .insert(proof.resource_address().clone().into());
+            }
+            for resource in &intent.auth_zone_init.simulate_every_proof_under_resources {
+                intent.references.insert(resource.clone().into());
+            }
         }
+
+        if let Some(root) = intents.get_mut(0) {
```

Comment:
Any reason why we restrict address allocation to intent 0?

Reply:
I'll add a comment